### ISSUE TITLE
Error with vector's key locally

### ISSUE THREAD

**Issue by PierreMarchand20** [cljoly/gohugo-asciinema#2]
The module seems to work fine when deployed, but for some reasons, locally I have this error:

<img width="1511" alt="image" src="https://user-images.githubusercontent.com/15056323/157719264-9f9247dd-0191-401e-98b7-d3372d2ea06b.png">

Any ideas why ?

By the way, I still have the error mentioned in #1 (locally and after deployment):

<img width="1510" alt="image" src="https://user-images.githubusercontent.com/15056323/157719501-f3b8d808-991d-407d-acd9-f4cacba5955a.png">

**Comment by cljoly** [comment]
Hi, sorry I missed this. I'll take a deeper look hopefully tomorrow.

**Comment by cljoly** [comment]
> The module seems to work fine when deployed, but for some reasons, locally I have this error:

Could you please share the exact shortcode code you are using in your markdown file when you get this error? I would imagine it looks something like this (for instance):
```
{{< asciicast src="/telescope-repo-nvim/telescope.json" poster="npt:0:04" autoplay="true" loop="true" >}}
```

**Comment by cljoly** [comment]
> By the way, I still have the error mentioned in #1 (locally and after deployment):

You get this from the master branch, right? It’s only fixed on the beta branch, I’ve edited https://github.com/cljoly/gohugo-asciinema/issues/1#issuecomment-1019236256 to make it clearer. I have also created #3 to consider backporting the fix to master.

**Comment by PierreMarchand20** [comment]
> > The module seems to work fine when deployed, but for some reasons, locally I have this error:
> 
> Could you please share the exact shortcode code you are using in your markdown file when you get this error? I would imagine it looks something like this (for instance):
> 
> ```
> {{< asciicast src="/telescope-repo-nvim/telescope.json" poster="npt:0:04" autoplay="true" loop="true" >}}
> ```

It is something like 
```
{{< asciicast src="/casts/2020_CTCW_2_git/setup.cast" rows="30" preload="1" theme="solarized-dark" title="Setup git">}}
```

**Comment by PierreMarchand20** [comment]
> > By the way, I still have the error mentioned in #1 (locally and after deployment):
> 
> You get this from the master branch, right? It’s only fixed on the beta branch, I’ve edited [#1 (comment)](https://github.com/cljoly/gohugo-asciinema/issues/1#issuecomment-1019236256) to make it clearer. I have also created #3 to consider backporting the fix to master.

I see, it may be better to use the beta branch then ? I am not sure how I can choose between both versions with hugo.

**Comment by cljoly** [comment]
Hey thanks for providing these details, I'll try to reproduce your error and document things some more this weekend.

**Comment by PierreMarchand20** [comment]
Thanks! and thank you for this module, it is indeed better than everyone doing its own thing (like I did on my webpage) :-)

**Comment by cljoly** [comment]
> I see, it may be better to use the beta branch then ? I am not sure how I can choose between both versions with hugo.

It’s a much smaller player in the new version, but it’s also considered beta by their developer and as such might be less stable so 🤷🏼. Given that you can easily go back and forth between the two versions, I think you can give it a try and see for yourself.

I’ve improved the doc in the readme of the [`3beta`](https://github.com/cljoly/gohugo-asciinema/tree/3beta) branch.

## Tldr:

To go to the beta version, run:
```
hugo mod get -u -v cj.rs/gohugo-asciinema@3beta
```
and to go back
```
hugo mod get -u -v cj.rs/gohugo-asciinema
```
(we are simply using the [go *version query suffix*](https://go.dev/ref/mod#go-get))

Does that make sense?

**Comment by cljoly** [comment]
Unfortunately, I couldn’t reproduce your error, neither with Chrome nor Firefox. I run
```
hugo serve
```

And I use
```
{{< asciicast src="/telescope.cast" rows="30" preload="1" theme="solarized-dark" title="Setup git">}}
```
with [this file](https://cj.rs/telescope-repo-nvim/telescope.json) (renamed).

**Comment by cljoly** [comment]
Maybe the issue is related to your particular asciicast?

**Comment by PierreMarchand20** [comment]
I actually retried after a update of safari (I should have said I am using safari...), and now I get this error:
<img width="1507" alt="image" src="https://user-images.githubusercontent.com/15056323/159155769-6d0638bc-f0b6-4624-bea0-f3b04e5842e8.png">

I guess I am missing something? And this happens with both versions.

**Comment by cljoly** [comment]
Hum that's weird

**Comment by PierreMarchand20** [comment]
I have the same issue with Firefox, could it be an issue with Hugo on my laptop? Like a wrong path or something, but it does find the module, otherwise I would have an error when calling `hugo serve`.

You can find my webpage here: https://github.com/PierreMarchand20/personal_webpage/tree/brouillon

**Comment by cljoly** [comment]
Ok, now I’ve got the error on your website (on page http://localhost:1313/post/2020_ctcw_2_git/ from branch brouillon)

![image](https://user-images.githubusercontent.com/7347374/159158985-05de18e7-4f95-4248-98e7-21b9b5b17234.png)

With Firefox and hugo version:
```
hugo v0.95.0+extended linux/amd64 BuildDate=unknown
```

**Comment by cljoly** [comment]
Turns out the scratchpad is not working as intended and the sources of the player don’t get inserted.

**Comment by cljoly** [comment]
Ok, so with `hugo serve` on your website, using Firefox:
* on the `master` branch of this repo, I get errors only if there are multiple shortcodes inserted in the page.
* on the `3beta` branch, it seems the js and css files of the player are included but not loaded.

**Comment by PierreMarchand20** [comment]
May be it is because of the theme I used? Before I had to add in the config file the js and css files, but I thought it was unnecessary now with Hugo modules.

**Comment by PierreMarchand20** [comment]
Here: https://wowchemy.com/docs/hugo-tutorials/extending-wowchemy/ do you agree that this kind of things should be unnecessary with Hugo modules ?

**Comment by cljoly** [comment]
> do you agree that this kind of things should be unnecessary with Hugo modules ?

Provided there is a suitable Hugo module for each third party script, I would think it’s unnecessary indeed.

**Comment by PierreMarchand20** [comment]
mmmm ok, so I do not see why the asciinema do not get loaded properly :-/

**Comment by PierreMarchand20** [comment]
And, as I said at the beginning, when deployed it works: https://deploy-preview-2--pierremarchand.netlify.app/post/2020_ctcw_2_git/

**Comment by cljoly** [comment]
Found another problem: on the beta version, somehow, an empty file is produced. There must be a problem with the way I produce the resource…

**Comment by cljoly** [comment]
> Found another problem: on the beta version, somehow, an empty file is produced. There must be a problem with the way I produce the resource…

And that does not happen on my personal website.

**Comment by cljoly** [comment]
Ah found the problem, on your `brouillon` branch, you have a file `assets/js/asciinema-player.min.js` that’s empty. I removed it and it worked. It was overwriting the asciinema JS provided by my module 😄 .

**Comment by cljoly** [comment]
I think I have a working version for you, let me refine it and make a PR to your website…

**Comment by PierreMarchand20** [comment]
Ah good catch! at some point I had some troubles to install this hugo module, so I tried to copy this file from somewhere...

**Comment by PierreMarchand20** [comment]
Deleting this file makes it working on my laptop, but I can let you do a PR (and close the issue).

**Comment by PierreMarchand20** [comment]
What is the issue with master branch of https://github.com/cljoly/gohugo-asciinema ? It seems to work on my laptop.

**Comment by PierreMarchand20** [comment]
(the player is a bit weird on safari with the beta of asciinema:)

<img width="779" alt="image" src="https://user-images.githubusercontent.com/15056323/159165244-88c3df85-253e-4253-a3db-367b74415cc2.png">

**Comment by cljoly** [comment]
> (the player is a bit weird on safari with the beta of asciinema:)
> 
> <img alt="image" width="779" src="https://user-images.githubusercontent.com/15056323/159165244-88c3df85-253e-4253-a3db-367b74415cc2.png">

With Firefox as well :)

**Comment by cljoly** [comment]
> What is the issue with master branch of https://github.com/cljoly/gohugo-asciinema ? It seems to work on my laptop.

I couldn’t get it to play the asciicast, but if it works, all good. Let’s close that issue then.

Thanks for taking the time to open that issue and for your cooperation debugging this!
